src/main/java/org/usfirst/frc/team4099/robot/subsystems/Drive.kt

```diff
@@ -30,16 +49,125 @@ class Drive private constructor() : Subsystem {
         rightSlave2SRX.set(ControlMode.Follower, Constants.Drive.RIGHT_MASTER_ID.toDouble())
         rightMasterSRX.set(ControlMode.PercentOutput, 0.0)
+        rightMasterSRX.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder, 0, 10)
+        rightMasterSRX.setSensorPhase(false)
+        rightMasterSRX.setStatusFramePeriod(StatusFrameEnhanced.Status_2_Feedback0, 5, 10)
+        rightMasterSRX.config_kP(0, Constants.Gains.RIGHT_LOW_KP, 10)
+        rightMasterSRX.config_kI(0, Constants.Gains.RIGHT_LOW_KI, 10)
+        rightMasterSRX.config_kD(0, Constants.Gains.RIGHT_LOW_KD, 10)
+        rightMasterSRX.config_kF(0, Constants.Gains.RIGHT_LOW_KF, 10)
+
+        rightMasterSRX.config_kP(1, Constants.Gains.RIGHT_HIGH_KP, 10)
+        rightMasterSRX.config_kI(1, Constants.Gains.RIGHT_HIGH_KI, 10)
+        rightMasterSRX.config_kD(1, Constants.Gains.RIGHT_HIGH_KD, 10)
+        rightMasterSRX.config_kF(1, Constants.Gains.RIGHT_HIGH_KF, 10)
+
+        leftMasterSRX.configVelocityMeasurementPeriod(VelocityMeasPeriod.Period_10Ms,10)
+        leftMasterSRX.configVelocityMeasurementWindow(32, 10)
+        rightMasterSRX.configVelocityMeasurementPeriod(VelocityMeasPeriod.Period_10Ms,10)
+        rightMasterSRX.configVelocityMeasurementWindow(32,10)
+
+        setOpenLoop(DriveSignal.NEUTRAL)
+
+
 
         ahrs = AHRS(SPI.Port.kMXP)
 
         this.zeroSensors()
     }
 
+
+    @Synchronized
+    fun setOpenLoop(signal: DriveSignal) {
+        if (currentState !== DriveControlState.OPEN_LOOP) {
+            leftMasterSRX.set(ControlMode.PercentOutput, 0.0)
+            rightMasterSRX.set(ControlMode.PercentOutput, 0.0)
+            leftMasterSRX.configNominalOutputForward(0.0, 10)
+            rightMasterSRX.configNominalOutputForward(0.0, 10)
+            currentState = DriveControlState.OPEN_LOOP
+            setBrakeMode(NeutralMode.Brake)
+        }
+        setLeftRightPower(signal.leftMotor, signal.rightMotor)
+    }
+
+
+
+
+    fun onStart(timestamp: Double){
+        synchronized (this) {
+            setOpenLoop(DriveSignal.NEUTRAL)
+            setBrakeMode(NeutralMode.Coast)
+            setVelocitySetpoint(0.0, 0.0) //could update in future
+        }
+    }
+
+    fun usesTalonVelocityControl(state: DriveControlState): Boolean {
+        if (state == DriveControlState.VELOCITY_SETPOINT || state == DriveControlState.PATH_FOLLOWING) {
+            return true
+        }
+        return false
+    }
+
+    fun usesTalonPositionControl(state: DriveControlState): Boolean {
+        if (state == DriveControlState.TURN_TO_HEADING) {
+            return true
+        }
+        return false
+    }
+
+
+    fun getHighGear():Boolean {
+        return highGear
+    }
+
+    fun setHighGear(choice: Boolean){
+        highGear=choice
+    }
+
+    fun getBrakeMode(): NeutralMode{
+        return brakeMode
+    }
+
+    fun setBrakeMode(type: NeutralMode){
+        if (brakeMode!=type) {
```

Comment:
I've missed this in a couple of places, but please use spaces between your binary operators. IntelliJ can do reformat code which should automatically fix all of them.